lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -175,19 +191,74 @@ data SelectionLimit
 
 -- | The result of performing a successful selection.
 --
-data SelectionResult = SelectionResult
+data SelectionResult change = SelectionResult
     { inputsSelected
         :: !(NonEmpty (TxIn, TxOut))
         -- ^ A (non-empty) list of inputs selected from 'utxoAvailable'.
+    , extraCoinSource
+        :: !(Maybe Coin)
+        -- ^ An optional extra source of ada.
+    , outputsCovered
+        :: ![TxOut]
+        -- ^ A list of ouputs covered.
+        -- FIXME: Left as a list to allow to work-around the limitation of
+        -- 'performSelection' which cannot run for no output targets (e.g. in
+        -- the context of a delegation transaction). This allows callers to
+        -- specify a dummy 'TxOut' as argument, and remove it later in the
+        -- result; Ideally, we want to handle this in a better way by allowing
+        -- 'performSelection' to work with empty output targets. At the moment
+        -- of writing these lines, I've already been yak-shaving for a while and
+        -- this is the last remaining obstacle, not worth the effort _yet_.
     , changeGenerated
-        :: !(NonEmpty TokenBundle)
+        :: !(NonEmpty change)
         -- ^ A (non-empty) list of generated change outputs.
     , utxoRemaining
         :: !UTxOIndex
         -- ^ The subset of 'utxoAvailable' that remains after performing
         -- the selection.
     }
-    deriving (Eq, Show)
+    deriving (Generic, Eq, Show)
+
+instance Buildable (SelectionResult TokenBundle) where
+    build = buildSelectionResult (blockListF . fmap TokenBundle.Flat)
+
+instance Buildable (SelectionResult TxOut) where
+    build = buildSelectionResult (blockListF . fmap build)
+
+buildSelectionResult
+    :: (NonEmpty change -> Builder)
+    -> SelectionResult change
+    -> Builder
+buildSelectionResult changeF s@SelectionResult{inputsSelected,extraCoinSource} =
+    mconcat
+        [ nameF "inputs selected"  (inputsF inputsSelected)
+        , nameF "extra coin input" (build extraCoinSource)
+        , nameF "outputs covered"  (build $ outputsCovered s)
+        , nameF "change generated" (changeF $ changeGenerated s)
+        , nameF "size utxo remaining" (build $ UTxOIndex.size $ utxoRemaining s)
```

Comment:
I agree is doesn't buy us much, but it also doesn't cost us much (literally, 2 white spaces). I think it make it still slightly more readable. Putting each one on a separate line seems excessive in that case.

Reply:
Hi @KtorZ I don't have a strong opinion on this, so will leave it to you to decide!

However, I do think we would benefit from:

1. Adhering to our existing coding standards as much as possible.
2. If we want to deviate from the standard, have a discussion in the channel about how we want to revise them.

As I'm sure no-one likes to have these discussions in PRs! 😹 

In this case, our existing coding standard says to avoid variable length indentation:
https://github.com/input-output-hk/adrestia/wiki/Coding-Standards#avoid-variable-length-indentation

I can completely understand the argument for aligning variable names in pattern matches, as it can aid with comprehension. But I'm not persuaded that vertical alignment is a good idea in the general case, for all the reasons pointed out in our standards document.

Perhaps we should discuss this in the channel, if we decide we want to revise our approach?